### PR TITLE
SMCP install config for 2.3

### PR DESCRIPTION
**Pull request by leoieggli**
Adding config to install SMCP version 2.3

### CODE REVIEW

File: pkg/ossm/smcp_install.go [hunk 1]
```diff
@@ -22,10 +22,10 @@ import (
 	"github.com/maistra/maistra-test-tool/pkg/util"
 )
 
-func installDefaultSMCP22() {
```
Comment (yxun) [inline pkg/ossm/smcp_install.go:25]: we still need the installDefaultSMCP22 functions in a 2.3 branch testing. 
We can copy that function and then apply your new version changes.

Comment (yxun) [reply pkg/ossm/smcp_install.go:25]: oh, wait. that's the default SMCP. Ignore my comment above.

File: pkg/ossm/smcp_install.go [hunk 2]
```diff
@@ -63,19 +63,19 @@ func TestSMCPInstall(t *testing.T) {
 		util.Shell(`oc get -n %s pods`, meshNamespace)
 	})
 
-	t.Run("smcp_test_uninstall_2.2", func(t *testing.T) {
+	t.Run("smcp_test_uninstall_2.3", func(t *testing.T) {
 		defer util.RecoverPanic(t)
-		util.Log.Info("Delete SMCP v2.2 in ", meshNamespace)
+		util.Log.Info("Delete SMCP v2.3 in ", meshNamespace)
 		util.KubeDeleteContents(meshNamespace, smmr)
 		util.KubeDeleteContents(meshNamespace, util.RunTemplate(smcpV22_template, smcp))
```
Comment (yxun) [inline pkg/ossm/smcp_install.go:85]: This uninstall 2.3 step needs to run smcpV23_template.

File: pkg/ossm/smcp_install.go [hunk 3]
```diff
@@ -94,17 +94,17 @@ func TestSMCPInstall(t *testing.T) {
...
 		time.Sleep(time.Duration(40) * time.Second)
 	})
 
-	t.Run("smcp_test_install_2.0", func(t *testing.T) {
+	t.Run("smcp_test_install_2.1", func(t *testing.T) {
 		defer util.RecoverPanic(t)
-		util.Log.Info("Create SMCP v2.0 in namespace ", meshNamespace)
+		util.Log.Info("Create SMCP v2.1 in namespace ", meshNamespace)
 		util.ShellMuteOutputError(`oc new-project %s`, meshNamespace)
 		util.KubeApplyContents(meshNamespace, util.RunTemplate(smcpV20_template, smcp))
```
Comment (yxun) [inline pkg/ossm/smcp_install.go:124]: This install 2.1 step needs to run smcpV21_template

File: pkg/ossm/smcp_install.go (hunk 1 above)
Comment (leoieggli) [reply pkg/ossm/smcp_install.go:25]: Added again

File: pkg/ossm/smcp_install.go (hunk 2 above)
Comment (leoieggli) [reply pkg/ossm/smcp_install.go]: done!

File: pkg/ossm/smcp_install.go (hunk 3 above)
Comment (leoieggli) [reply pkg/ossm/smcp_install.go]: done!

**Review by yxun — APPROVED** [review]

**Comment by mattmahoneyrh** [comment]
LGTM. Prior to merge, please do a test run of this PR against OSSM 2.3.

**Comment by leoieggli** [comment]
Local tests passed!

`servicemeshcontrolplane.maistra.io/basic condition met
--- PASS: A1 (399.92s)
    --- PASS: A1/smcp_test_install_2.3 (36.64s)
    --- PASS: A1/smcp_test_uninstall_2.3 (42.93s)
    --- PASS: A1/smcp_test_install_2.2 (26.39s)
    --- PASS: A1/smcp_test_uninstall_2.2 (42.93s)
    --- PASS: A1/smcp_test_install_2.1 (68.10s)
    --- PASS: A1/smcp_test_uninstall_2.1 (63.10s)
    --- PASS: A1/smcp_test_upgrade_2.1_to_2.2 (57.04s)
    --- PASS: A1/smcp_test_upgrade_2.2_to_2.3 (38.88s)
PASS
ok  	github.com/maistra/maistra-test-tool/tests	451.979s`

**Pull request merged**